great_expectations/agent/actions/run_onboarding_data_assistant.py

```diff
@@ -0,0 +1,86 @@
+from typing import TYPE_CHECKING
+
+from great_expectations.agent.actions.action import (
+    ActionResult,
+    AgentAction,
+    CreatedResource,
+)
+from great_expectations.agent.models import RunOnboardingDataAssistantEvent
+from great_expectations.exceptions import StoreBackendError
+
+if TYPE_CHECKING:
+    from great_expectations.core import ExpectationSuite
+
+
+class RunOnboardingDataAssistantAction(AgentAction):
+    def run(self, event: RunOnboardingDataAssistantEvent, id: str) -> ActionResult:
+        expectation_suite_name = f"{event.data_asset_name} onboarding assistant suite"
+        checkpoint_name = f"{event.data_asset_name} onboarding assistant checkpoint"
+
+        # ensure resources we create don't already exist
+        try:
+            self._context.get_expectation_suite(
+                expectation_suite_name=expectation_suite_name
+            )
+            raise ValueError(
+                f"Onboarding Assistant Expectation Suite `{expectation_suite_name}` already exists. "
+                + "Please rename or delete suite and try again"
+            )
+        except StoreBackendError:
+            # resource is unique
+            pass
+
+        try:
+            self._context.get_checkpoint(name=checkpoint_name)
+            raise ValueError(
+                f"Onboarding Assistant Checkpoint `{checkpoint_name}` already exists. "
+                + "Please rename or delete Checkpoint and try again"
+            )
+        except StoreBackendError:
+            # resource is unique
+            pass
+
+        datasource = self._context.get_datasource(datasource_name=event.datasource_name)
+        asset = datasource.get_asset(asset_name=event.data_asset_name)
+        batch_request = asset.build_batch_request()
+
+        data_assistant_result = self._context.assistants.onboarding.run(
+            batch_request=batch_request,
+        )
```

Comment:
This is where 99% of the effort goes right? There's a lot of complexity here - should we wrap with a `try/except` to cover possible failure cases?

Could be fine as is.

Reply:
if there isn't a path to retry, then raising an exception here is appropriate. This code is running in a thread, so if the job fails we need the thread to exit. Reporting the failure is the responsibility of the caller, currently `GXAgent`